### PR TITLE
Only run test features on AGP7 / Gradle 7

### PR DESCRIPTION
**Pull request by lemnik**
## Goal
Avoid running test features on earlier versions of AGP / Gradle as they won't pass due to the compatibility checks.

This PR also disables all of our NDK tests due to a change in where AGP places it's intermediate `so` files. These tests will be re-enabled once we have a fix in-place for that change. 

## Testing
Testing relies on existing tests to illustrate the plugin compatibility. The only changes are to comment-out the NDK related tests (as mentioned above).

### CODE REVIEW

File: features/proxy.feature
Comment (fractalwrench) [inline features/proxy.feature:42]: These tags can probably be removed throughout the tests and where they're defined in the Ruby file.

File: .travis.yml
Comment (fractalwrench) [inline .travis.yml:97]: It might be nice to define these explicitly just so that it's obvious what is being used, and to make it easy to change in future from one place.

File: .travis.yml
Comment (fractalwrench) [inline .travis.yml:56]: This is an existing problem, but I think it might be worth tweaking this step so that it's combined with the unit test job, as it'll skip 9 minutes of what is mostly environment setup on CI.

**Review by fractalwrench — COMMENTED** [review]
These changes look good, I left a few comments inline.

I think we will need to fix the license checks before merging this work - so it'd be good to get an idea of exactly how it's failing and what would be required to get it compatible with Gradle 7. Depending on the complexity we could look into forking it ourselves, finding a similar plugin, or rolling our own license check solution.

File: .travis.yml
Comment (lemnik) [reply .travis.yml:97]: We do also (sort of) have this in `env.rb`: https://github.com/bugsnag/bugsnag-android-gradle-plugin/pull/395/files#diff-3c1d97c5f1fa7ca90a4a96df18478ce19fe4d3a8dbaa3914ccc6ec6c43e81cbe

**Review by fractalwrench — APPROVED** [review]
LGTM pending CI

**Pull request merged**
